src/rapids_dependency_file_generator/rapids_dependency_file_generator.py

```diff
@@ -112,8 +112,8 @@ def should_use_specific_entry(matrix_combo, specific_entry_matrix):
     return True
 
 
-def main(config_file, files):
-    with open(config_file, "r") as f:
+def main(config_file_path, files={}):
```

Comment:
Avoid mutable defaults. Use `None` instead of `{}` and then write
```python
if files is None:
    files = {}
```
(This blog post may be a little strongly worded but it explains the issues that can arise.)
https://florimond.dev/en/posts/2018/08/python-mutable-defaults-are-the-source-of-all-evil/